test/unit/plugins/rest_spec.js

```diff
@@ -821,20 +948,80 @@ describe("The Rest plugin", function () {
 				return instances.getInstance({ id : instance.id });
 			})
 			.then(function (instance) {
-				updatedInstance = new Instance({
-					id       : instance.id,
+				var request = new Request("PUT", "/v1/instances/" + instance.id).mime("application/json")
+				.payload({
 					ami      : instance.ami,
 					type     : instance.type,
-					state    : "failed",
 					uri      : instance.uri,
-					revision : instance.revision + 1
+					state    : "terminating",
+					revision : instance.revision
 				});
+				return request.inject(server);
+			})
+			.then(function (response) {
+				result = response.result;
+			});
+		});
+
+		after(function () {
+			updateStub.restore();
+		});
+
+		it("throws a 500 error", function () {
+			expect(result.statusCode).to.equal(500);
+		});
+	});
+	describe("when the database fails after terminateInstances has run", function () {
+
+		var result;
+		var updateStub;
+		var terminateInstancesStub;
+		var server     = new Hapi.Server();
+		var instances  = new InstanceAdapter();
+		var awsAdapter = new AwsAdapter();
+		before(function () {
+
+			return instances.createInstance()
+			.then(function (instance) {
+				updateStub = Sinon.stub(instances, "updateInstance").returns(
+					Bluebird.resolve({
+						id       : instance.id,
+						ami      : instance.ami,
+						type     : instance.type,
+						state    : "terminating",
+						uri      : instance.uri,
+						revision : instance.revision + 1
+					})
+				);
+				updateStub.onCall(1).rejects(new Error("Connection to the database fails."));
```

Comment:
We still need some way to verify that this is executed before we run the assertion. Otherwise, it may be that the instance is still in the "terminating" state simply because the async update hasn't actually happened yet.